### DISCUSSION TITLE
Mathematics operator overloading

### DISCUSSION THREAD

**Discussion by LukyGuyLucky** [c3lang/c3c#1362] [Q&A] [answered]
How can one implement a UDT with mathematic operator overloading or the like,for example,a Complex type with operator+/-/*/\ ... etc supported in c3?Or the language just doesn't support,the only way is to implement by functions/methods?

Thanks in advance.

**Comment by lerno** [comment]
The only way is using methods at this time.

**Comment by lerno** [comment]
0.7.1 will have general arithmetics overloading. The complex type uses it too.

**Reply by LukyGuyLucky** [reply, marked as answer]
Thanks for letting me know.
